tsdb/engine/tsm1/encoding.go

```diff
@@ -220,11 +239,13 @@ func encodeFloatBlock(buf []byte, values []Value) ([]byte, error) {
 	// for timestamps and values.
 
 	// Encode values using Gorilla float compression
-	venc := NewFloatEncoder()
+	venc := getFloatEncoder()
+	defer putFloatEncoder(venc)
```

Comment:
It was probably a little vague, but I meant to put back the float encoder in the error block for the time encoder...

This is what I'm suggesting:
```go

tb, err := tsenc.Bytes()
putTimeEncoder(tsenc)
if err != nil {
    putFloatEncoder(venc)
    return nil, err
}

vb, err := venc.Bytes()
putFloatEncoder(venc)
if err != nil {
    return nil, err
}
```